src/pymor/algorithms/scm.py

```diff
@@ -0,0 +1,245 @@
+        evals_ub = np.array([self.ub_functional.evaluate(mu) for mu in mus])
+        evals_lb = np.array([self.lb_functional.evaluate(mu) for mu in mus])
+        estimated_errors = (evals_ub - evals_lb) / evals_ub
+        if return_all_values:
+            return estimated_errors
+        else:
+            index_max_error = np.argmax(estimated_errors)
+            return estimated_errors[index_max_error], mus[index_max_error]
+
+    def extend(self, mu):
+        self.constraint_parameters.append(mu)
+        fixed_parameter_op = self.operator.assemble(mu)
+        eigvals, eigvecs = eigs(fixed_parameter_op, k=1, which='SM', E=self.product)
+        self.coercivity_constants.append(eigvals[0].real)
+        minimizer = eigvecs[0]
+        minimizer_squared_norm = minimizer.norm(product=self.product) ** 2
+        y_opt = np.array([op.apply2(minimizer, minimizer)[0].real / minimizer_squared_norm
+                          for op in self.operator.operators])
+        self.minimizers.append(y_opt)
+        self.lb_functional = LBSuccessiveConstraintsFunctional(self.operator, self.constraint_parameters,
+                                                               self.coercivity_constants, self.bounds,
+                                                               **self.params_lb_functional)
+        self.ub_functional = UBSuccessiveConstraintsFunctional(self.operator, self.constraint_parameters,
+                                                               self.minimizers, **self.params_ub_functional)
+
+
+def construct_scm_functionals(operator, training_set, initial_parameter, atol=None, rtol=None, max_extensions=None,
+                              product=None, params_lb_functional={}, params_ub_functional={}):
+    """Method to construct lower and upper bounds using the successive constraints method.
+
+    Parameters
+    ----------
+    operator
+        |LincombOperator| for which to provide a bounds on the
+        coercivity constant.
+    training_set
+        |Parameters| used as training set for the greedy algorithm.
+    initial_parameter
+        |Parameter| used to initialize the surrogate for the greedy algorithm.
+    atol
+        If not `None`, stop the greedy algorithm if the maximum (estimated)
+        error on the training set drops below this value.
+    rtol
+        If not `None`, stop the greedy algorithm if the maximum (estimated)
+        relative error on the training set drops below this value.
+    max_extensions
+        If not `None`, stop the greedy algorithm after `max_extensions`
+        extension steps.
+    product
+        Product with respect to which the coercivity constant should be
+        estimated.
+    params_lb_functional
+        Additional parameters passed to the constructor of the functional
+        for the lower bound.
+    params_ub_functional
+        Additional parameters passed to the constructor of the functional
+        for the upper bound.
```

Comment:
I'm confused. For the UB functional, there should be no additional parameters to pass? For the LB functional, these should only be `linprog_method`, `linprog_options` and `M`, right?  Maybe it is better to make them explicit here, as the user probably never will directly touch the functionals.

Reply:
I added the parameters explicitly now.